### ISSUE TITLE
Setup prod hosting of static media for AWS prod

### ISSUE THREAD

**Issue by pmac** [mozilla/kitsune#2949]
For dev we'll probably just serve static media from the primary docker container via [whitenoise](http://whitenoise.evans.io/en/stable/), but that won't scale to prod traffic. We have some options:

* Sync static files to S3 and serve via a CDN
* Deploy a separate Nginx container for serving said files, optionally with a CDN in front
* Put a CDN in front of the entire site (potentially a lot more work)

**Comment by glogiotatidis** [comment]
> Deploy a separate Nginx container for serving said files, optionally with a CDN in front

In this case we can also build a single image that will either run `nginx-light` or `gunicorn` depending on the value of a env variable. This would simply the building pipeline, keep the statics with the code, while we keep the flexibility to scale each function (static, app) seperately.

**Comment by metadave** [comment]
regarding S3 / Cloudfront: https://github.com/mozmeao/infra/issues/391

**Comment by glogiotatidis** [comment]
I suggest that we go fwd with the following plan:
 
  1. Enable WhiteNoise for all dev, stage, prod
  2. Setup CDN to serve only static files and not the whole site. See [instructions](http://whitenoise.evans.io/en/stable/django.html#restricting-cloudfront-to-static-files) from WhiteNoise docs.

This way:
 - Statics remain with the code and we build only one docker image
 - We don't have to manage S3 headers and deal with other S3 oddities
 - We don't maintain nginx deployments
 - Avoid the work and problems related to putting the entire site behind CDN.
 - We maintain current SEO as the content will be only accessible through support.m.o and not the CDN url.

To move forward we need:
 - [ ] 3 CDNs (dev, stage, prod) configured as described in WhiteNoise docs.
 - [ ] DNS entries for the CDNs above. Names `static-media-{dev, stage, prod}-cdn.sumo.mozilla.net` CNAME to `static-media-{dev,stage,prod}-cdn.sumo.moz.works`
 - [ ] Switch to Whitenoise Storage backend ([code](https://github.com/mozilla/kitsune/blob/master/kitsune/settings.py#L383))
 - [ ] Bonus points by enabling [Brotli compression](http://whitenoise.evans.io/en/stable/django.html#brotli-compression) (docker images need update to include packages)

What do you think @mozilla/sumo?

**Comment by metadave** [comment]
How do these CDN setting look? https://github.com/mozmeao/infra/pull/694

**Comment by glogiotatidis** [comment]
> Switch to Whitenoise Storage backend (code)
> Bonus points by enabling Brotli compression (docker images need update to include packages)

We're currently using Pipeline which takes care of static files collection *and* generates bundles with the assets. Pipeline works OK and the bundles are useful and in addition to what whitenoise offers. So we keep the Pipeline based Storage backend as we have it now.

I tested the current Whitenoise implementation and it  sets proper headers for static files
`Cache-Control: max-age=315360000, public, immutable`.

tl;dr no changes needed on the code side to move fwd with this. We can evaluate replacing Pipeline with something different / more modern in the future.

**Comment by metadave** [comment]
> Setup CDN to serve only static files and not the whole site. See instructions from WhiteNoise docs.

per the [WhiteNoise docs](http://whitenoise.evans.io/en/stable/django.html#restricting-cloudfront-to-static-files), each CDN is limited to `static/*`.

(See the Behaviors tab for each CDN below)
- [dev](https://console.aws.amazon.com/cloudfront/home?region=us-west-2#distribution-settings:EQZG8RJ5XZQYN)
- [prod](https://console.aws.amazon.com/cloudfront/home?region=us-west-2#distribution-settings:E37EUKSZPJXELT)
- [stage](https://console.aws.amazon.com/cloudfront/home?region=us-west-2#distribution-settings:E1QQZ0CM2MO9S3)

**Comment by glogiotatidis** [comment]
I did setup sumo-dev.frankfurt.moz.works to use dev CDN and it works like a charm. nice work as always @metadave 

I guess all we need now is the `mozilla.net` dns entries.

**Comment by metadave** [comment]
Sorry, should have dropped the bug # in when I created it last week:
https://bugzilla.mozilla.org/show_bug.cgi?id=1433561

**Comment by glogiotatidis** [comment]
This is now done. Update from bug

```
CNAMEs created.

FQDN 		Target 	
static-media-dev-cdn.sumo.mozilla.net 	CNAME 	static-media-dev-cdn.sumo.moz.works


FQDN 		Target 	
static-media-stage-cdn.sumo.mozilla.net 	CNAME 	static-media-stage-cdn.sumo.moz.works


FQDN 		Target 	
static-media-prod-cdn.sumo.mozilla.net 	CNAME 	static-media-prod-cdn.sumo.moz.works```
